### PR TITLE
Extend build-in-pause-mode to entrances and exits, clear up some flags, fix ghost sticking in pause mode, fixes #1918

### LINKED ISSUE #1918: Entrance/exit ghosts are not removed when game is in pause mode
To reproduce:
- Build a ride.
- Turn pause mode on.
- Try placing an entrance or exit. It will place ghosts but not remove them.

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/ride/ride.c [hunk 1]
```diff
@@ -6697,8 +6697,8 @@ void game_command_place_ride_entrance_or_exit(int *eax, int *ebx, int *ecx, int
 money32 remove_ride_entrance_or_exit(sint16 x, sint16 y, uint8 rideIndex, uint8 station_num, uint8 flags){
 	rct_ride* ride = GET_RIDE(rideIndex);
 
-	if (!(flags & (1 << 6))){
-		if (RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0){
+	if (!(flags & GAME_COMMAND_FLAG_GHOST)){
```
Comment (janisozaur) [inline src/ride/ride.c]: You could fold the two checks into
`!(flags & (GAME_COMMAND_FLAG_GHOST | GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED))`

Comment (Gymnasiast) [inline src/ride/ride.c]: You're right. I don't do that because place_ride_entrance_or_exit did it like that, but that can be changed as well of course.

Comment (IntelOrca) [inline src/ride/ride.c]: There is usually no advantage to this as the compiler should optimise it out.

Comment (janisozaur) [inline src/ride/ride.c]: True, this is to be expected when optimization is enabled, but it makes the condition more concise, thus slightly shorter line and easier to fit in head.

Comment (Gymnasiast) [inline src/ride/ride.c]: @IntelOrca : Would you merge this if there's no advantage to changing?

**Pull request merged**